web/docs/auth/AuthPills.jsx

```diff
@@ -0,0 +1,50 @@
+import React from "react";
+import Link from "@docusaurus/Link";
+import './AuthPills.css'
+
+const PILL_INFO = {
+  email: { label: "Email", link: "/docs/auth/email" },
+  username: {
+    label: "Username & Password",
+    link: "/docs/auth/username-and-pass",
+  },
+  slack: { label: "Slack", link: "/docs/auth/social-auth/slack" },
+  discord: { label: "Discord", link: "/docs/auth/social-auth/discord" },
+  github: { label: "Github", link: "/docs/auth/social-auth/github" },
+  google: { label: "Google", link: "/docs/auth/social-auth/google" },
+  keycloak: { label: "Keycloak", link: "/docs/auth/social-auth/keycloak" },
+};
+
+function Pill({ label, link }) {
+  return (
+    <Link
+      to={link}
+      className={'pills'}
+    >
+      {label}
+    </Link>
+  );
+}
+
+export function AuthPills({ providers }) {
+  const validPills = providers
+    .map((provider) => PILL_INFO[provider])
+    .filter(Boolean);
```

Comment:
From #2818

> Let's make this a bit more robust and resistant to mis-use: instead of silently ignoring bad provider names, throw an error if the provider name doesn't exist!